backend.native/tests/interop/objc/tests/kt41811.kt

```diff
@@ -81,3 +81,29 @@ private fun createGarbageDeallocLoadWeak(weakRef: WeakReferenceProtocol) {
         assertSame(obj, weakDeallocLoadWeak!!.referent)
     }
 }
+
+@Test
+fun testKT41811WithGlobal() {
+    // Attempt to make the state predictable:
+    kotlin.native.internal.GC.collect()
+
+    deallocRetainReleaseDeallocated = false
+    assertFalse(deallocRetainReleaseDeallocated)
+
+    autoreleasepool {
+        {
+            globalDeallocRetainRelease = object: DeallocRetainRelease() {}
+        }()
+        // Clean up local DeallocRetainRelease on Kotlin side
+        kotlin.native.internal.GC.collect()
+        // And drop the last reference to DeallocRetainRelease from ObjC global scope.
+        globalDeallocRetainRelease = null
+    }
+
+    assertFalse(deallocRetainReleaseDeallocated)
+
+    // This will dispose DeallocRetainRelease on Kotlin side, which will cause dealloc
+    // on ObjC side, which will immediately schedule another disposal on the Kotlin side.
+    kotlin.native.internal.GC.collect()
```

Comment:
Crashing occurs because `freeContainer` ran `scheduleDestroyContainer` which runs `setNextLink`: https://github.com/JetBrains/kotlin-native/blob/0a5e64f6175f63db833e32e88df2e5c715ee8fc2/runtime/src/main/cpp/Memory.h#L282-L284
which pretty much overwrites `ObjHeader`; so the next time someone accesses the object (e.g. checks its type info), it contains garbage.

Other tests don't crash, because during their run of `freeContainer` `scheduleDestroyContainer` isn't called: https://github.com/JetBrains/kotlin-native/blob/0a5e64f6175f63db833e32e88df2e5c715ee8fc2/runtime/src/main/cpp/Memory.cpp#L1108-L1109 Here buffered flag is reset because of `GC.collect()` at line 98.